src/pip/_internal/metadata/_json.py

```diff
@@ -64,7 +64,8 @@ def sanitise_header(h: Union[Header, str]) -> str:
         key = json_name(field)
         if multi:
             value: Union[str, List[str]] = [
-                sanitise_header(v) for v in msg.get_all(field)  # type: ignore
+                sanitise_header(v)
+                for v in msg.get_all(field)  # type: ignore
```

Comment:
This may be a semantic change. If the `# type: ignore` directive is per-line (rather than per-expression) then the `sanitise_header(v)` will now be type-checked.

Reply:
mypy is happy with it, so it should be fine

Reply:
Seems this is an implementation based semantic (i.e. no standard), for MyPy you should be getting a consistent behavior on Python 3.8+ (but potentially not if you were to run it on 3.7, which Pip's CI doesn't do): https://github.com/python/mypy/pull/6648

Reply:
You may have misunderstood my comment (or I may have misunderstood your reply). From [PEP 484](https://peps.python.org/pep-0484/#type-comments), "The `# type: ignore` comment should be put on the line that the error refers to". This suggests that if there *were* an error in line 67 (`sanitise_header(v)`), the ignore comment wouldn't apply to it now, because that ignore is on line 68.

I know I'm being very pedantic here, but I'm bothered by the general idea that ruff is failing to reformat in a way that's semantically identical.

The case in `test_utils.py` below is even more problematic, IMO.